libraries/tools/kotlin-maven-plugin/src/main/java/org/jetbrains/kotlin/maven/K2JSCompilerMojo.java

```diff
@@ -86,7 +86,8 @@ protected void configureSpecificCompilerArguments(@NotNull K2JSCompilerArguments
             collector.add(new File(outputFile).getParent());
         }
         if (metaInfo) {
-            String metaFile = KotlinPackage.substringBeforeLast(outputFile, JavaScript.DOT_EXTENSION, outputFile) + KotlinJavascriptMetadataUtils.META_JS_SUFFIX;
+            String output = com.google.common.base.Objects.firstNonNull(outputFile, ""); // fqname here because of J8 compatibility issues
```

Comment:
@udalov we don't have exact thing in stdlib because there is elvis operator in K that covers this functionality